source/common/stats/thread_local_store.cc

```diff
@@ -75,6 +91,36 @@ void ThreadLocalStoreImpl::shutdownThreading() {
   shutting_down_ = true;
 }
 
+void ThreadLocalStoreImpl::mergeHistograms() {
+  if (!shutting_down_) {
+    std::unique_lock<std::mutex> lock(lock_);
+    // TODO: Wait till all threads have been checked-in
```

Comment:
@mattklein123 I need your help here. How do I wait till all worker threads have finished? Is there a standard pattern in c++/envoy code here that I can use? Tried to look at latch/barrier support but could not find such thing. This is the only thing pending from basic implementation.

Reply:
In a perfect world you could add barrier/notify functionality into the TLS itself, e.g., `runOnAllThreadsBarrier(...)`. If you don't want to do that, I would just add a function to TLS to get the number of workers, then in the run on all threads lambda, decrement an atomic counter. Whichever wins the race to 0 can post back to the main thread.

Reply:
Added the method in TLS system.